doc/nrf/ug_zigbee_configuring_zboss_traces.rst

```diff
@@ -0,0 +1,149 @@
+
+
+.. note::
+     ZBOSS trace logs are meant to be used for debugging ZBOSS based applications.
+     These libraries are not certified and should not be used for production firmware or end products.
+
+
+
+Requirements
+************
+
+To collect and view the log files, you will need software that can do the following:
+
+* Allow for reading serial ports
+  For example, if you are using Windows you can use PuTTY. (See :ref:`putty` for more information.)
+
+* Find the correct COM port
+  One way do this is by setting the Kconfig option :kconfig:`CONFIG_USB_DEVICE_PRODUCT="Zigbee sample"` to help identify the device in the system's :guilabel:`Device Manager`.
+
+
+.. rst-class:: numbered-step
+
+Switch to ZBOSS libraries with compiled-in trace logs
+*****************************************************
+
+Set the Kconfig option :kconfig:`CONFIG_ZIGBEE_ENABLE_TRACES` to switch to ZBOSS libraries with compiled-in trace logs.
+
+The ZBOSS stack comes in a precompiled form and trace logs are not compiled-in by default.
+An additional set of ZBOSS libraries are available in nrfxlib, which does have trace logs compiled-in.
+
+
+.. rst-class:: numbered-step
+
+Select which ZBOSS trace logs to print
+**************************************
+
+1. Select which subsystems to receive logs from by configuring the ZBOSS trace mask with Kconfig option :kconfig:`CONFIG_ZBOSS_TRACE_MASK`.
+   Trace masks can be created by adding masks of subsystems to receive trace logs from.
+   For available subsystems, see :file:`zb_trace.h`. (NOTE: CHANGE THIS TO PATH TO FILE, instead of just file name!)
+
+#. Select the level of logs you want to receive. Configure ZBOSS trace level by selecting one of the following levels:
+
+   * Trace logs turned off - set :kconfig:`CONFIG_ZBOSS_TRACE_LOG_LEVEL_OFF`
+   * Error trace logs level - set :kconfig:`CONFIG_ZBOSS_TRACE_LOG_LEVEL_ERR`
+   * Warning trace logs level - set :kconfig:`CONFIG_ZBOSS_TRACE_LOG_LEVEL_WRN`
+   * Info trace logs level - set :kconfig:`CONFIG_ZBOSS_TRACE_LOG_LEVEL_INF`
+   * Debug trace logs level - set :kconfig:`CONFIG_ZBOSS_TRACE_LOG_LEVEL_DBG`
+
+See :ref:`zigbee_ug_logging_stack_logs` to read more infomration about trace logs.
+
+
+
+.. rst-class:: numbered-step
+
+Configure how to print ZBOSS trace logs
+***************************************
+
+The :ref:`zigbee_osif_zboss_osif_serial` offers a few backends to choose from for printing ZBOSS trace logs.
+It is recommended to use the Zigbee serial logger, as it is the most efficient.
+To enable it, set the Kconfig option :kconfig:`CONFIG_ZBOSS_TRACE_BINARY_LOGGING`.
+
+**Optional:** You can increase size of the ring buffer that temporarily stores the trace logs.
+   * To do this, set the Kconfig option :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_BUFFER_SIZE`.
+   * This can prevent losing some of the logs in demanding scenarios. --NOTE: LOOK INTO THIS - GIVE A SPECIFIC EXAMPLE?
```

Comment:
Is there something we can add to this @sebastiandraus? Like a quick example, or another phrasing just so it is clear to the reader what a demanding scenario might be.